[#13]feat:Redis session storage 연동

Yo0oN opened the pull request:
Redis 연동

[#13]Redis session storage 연동

Yo0oN committed:
feat:Redis session storage 연동

Redis 연동

[#13]Redis session storage 연동

f-lab-dev commented on src/main/java/com/festa/config/RedisConfig.java:
```diff
@@ -0,0 +1,47 @@
+    private static int redisPort;
+
+    /*
+    RedisConnectionFactory : Redis와 연결
```
`Redis와 연결`이라는 말은 너무 단순한듯 한데 이 클래스의 역할은 정확하게 무엇일까요?

Yo0oN replied:
Thread-safe한 레디스 connection을 만들어 주는 클래스입니다.

f-lab-dev replied:
어떤 점 때문에 `thread-safe` 할까요?

Yo0oN replied:
RedisConnectionFactory는 레디스와 연결하는 일을 추상화 한 인터페이스입니다. 해당 인터페이스를 구현하여  레디스와 연결을 해주는 커넥터 역할을 하는 클래스가 있는데, 대표적으로 JedisConnectionFactory와 LettuceConnectionFactory입니다.
Jedis는 사용하기 쉽지만, 멀티스레드 환경을 이용하기 위해선 커넥션 풀이 필요하고 thread-safe하지 않습니다. 또, 풀에 연결된 수가 늘어나면 응답 시간이 오래걸립니다. 만약 풀을 이용하지 않는다면 timeout이 발생합니다.
Lettuce는 netty기반 클라이언트로 멀티스레드 환경에서 thread-safe합니다. 또 Netty는 비동기 NIO를 사용하며, Jedis보다 빠른 속도를 보입니다.
스프링부트에서는 기본 의존성으로 lettuce를 사용하고 있어 thread-safe 합니다.

f-lab-dev commented on src/main/java/com/festa/config/RedisConfig.java:
```diff
@@ -0,0 +1,47 @@
+    }
+
+    /*
+    RedisTemplate : 객체 직렬화를 해줌
```
`RedisTemplate`의 역할이 직렬화일까요??

이런 것은 도큐먼트를 찾아보면 알아볼 수 있습니다~

https://docs.spring.io/spring-data/redis/docs/current/api/org/springframework/data/redis/core/RedisTemplate.html

Yo0oN replied:
레디스 데이터 접근 코드를 간단하게 해주는 클래스입니다.
객체를 자동으로 직렬화, 역직렬화 해줍니다.

f-lab-dev replied:
`직렬화, 역직렬화`를 해주는 책임은 `RedisTemplate`의 책임일까요?

Yo0oN replied:
RedisTemplate 의 책임은 레디스 데이터에 쉽게 접근하는데 도움을 주는것입니다.

f-lab-dev commented:
의미를 알게되셨으니 주석도 바꿔주시는게 좋지 않을까요?

Yo0oN committed:
[#13]feat:Redis session storage 연동

RedisConnectionFactory, RedisTemplate애 대한 설명 수정

[#13]feat:Redis session storage 연동

Yo0oN commented:
> 의미를 알게되셨으니 주석도 바꿔주시는게 좋지 않을까요?

넵 수정하였습니다!!
